lib/client/local_proxy_middleware.go

```diff
@@ -57,11 +57,52 @@ func NewCertChecker(certIssuer CertIssuer, clock clockwork.Clock) *CertChecker {
 	}
 }
 
-// Create a new CertChecker for the given database.
+// Create a new CertChecker for the given database and teleport client.
 func NewDBCertChecker(tc *TeleportClient, dbRoute tlsca.RouteToDatabase, clock clockwork.Clock) *CertChecker {
 	return NewCertChecker(&DBCertIssuer{
-		Client:     tc,
-		RouteToApp: dbRoute,
+		IssueCertFunc: func(ctx context.Context) (tls.Certificate, error) {
```

Comment:
nit: would be easier to read if putting this as a separate function instead of nesting here. I think nesting is fine when the code is small and not many levels.